pkg/paravirt/paravirt_protected_entity_type_manager.go

```diff
@@ -166,3 +186,230 @@ func (this *ParaVirtProtectedEntityTypeManager) getDataTransports(id astrolabe.P
 
 	return data, md, combined, nil
 }
+
+// CreateFromMetadata creates CloneFromSnapshot CR in the Supervisor Cluster
+func (this *ParaVirtProtectedEntityTypeManager) CreateFromMetadata(ctx context.Context, metadata []byte, sourceSnapshotID astrolabe.ProtectedEntityID, componentSourcePETM astrolabe.ProtectedEntityTypeManager, cloneFromSnapshotNamespace string, cloneFromSnapshotName string, backupRepositoryName string) (astrolabe.ProtectedEntity, error) {
+	this.logger.Infof("CreateFromMeta called on Paravirtualized PETM. sourceSnapshotID: %s. cloneFromSnapshot: %s/%s, backupRepositoryName: %s", sourceSnapshotID.String(), cloneFromSnapshotNamespace, cloneFromSnapshotName, backupRepositoryName)
+
+	// Get backupRepository from Guest, set backRepositoryName to backupRepositoryObj.SvcBackupRepositoryName
+	var backupRepo *snapshotUtils.BackupRepository
+	if backupRepositoryName != "" {
```

Comment:
do we need to check `backupRepositoryName != utils.WithoutBackupRepository` here?

Reply:
Sure.  Will add that.